front/formaccesstype.form.php

```diff
@@ -0,0 +1,99 @@
+ * LICENSE
+ *
+ * This file is part of Formcreator.
+ *
+ * Formcreator is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Formcreator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Formcreator. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ * @copyright Copyright © 2011 - 2021 Teclib'
+ * @license   http://www.gnu.org/licenses/gpl.txt GPLv3+
+ * @link      https://github.com/pluginsGLPI/formcreator/
+ * @link      https://pluginsglpi.github.io/formcreator/
+ * @link      http://plugins.glpi-project.org/#/plugin/formcreator
+ * ---------------------------------------------------------------------
+ */
+
+include ('../../../inc/includes.php');
+
+Session::checkRight('entity', UPDATE);
+
+// Check if plugin is activated...
+if (!(new Plugin())->isActivated('formcreator')) {
+   Html::displayNotFoundError();
+}
+
+// Get target form
+$form_id = $_POST[PluginFormcreatorForm::getForeignKeyField()] ?? null;
+if (is_null($form_id)) {
+   http_response_code(400);
+   die;
+}
+
+$form = PluginFormcreatorForm::getById($form_id);
+if (!$form) {
+   Html::displayNotFoundError();
+}
+
+// No update if `access_rights` is not modified, keeping the save behavior as
+// the previous form_profile.form.php file
+if (!isset($_POST['access_rights'])) {
+   Html::back();
+   die;
+}
+
+// Prepare input
+$input = [
+   'id'                 => (int) $form_id,
+   'is_captcha_enabled' => $_POST['is_captcha_enabled'] ?? false,
+   'access_rights'      => (int) $_POST['access_rights'],
+   'users'              => [],
+   'groups'             => [],
+   'profiles'           => [],
+   'entities'           => [],
+];
+
+// Parse users/groups/profiles
+// $_POST['restrictions'] contains a flattened multi itemtype dropdown like so:
+// [
+//    0 => 'users_id-3',
+//    1 => 'users_id-14',
+//    2 => 'groups_id-2',
+//    3 => 'groups_id-78',
+//    4 => 'profiles_id-1',
+// ]
+foreach ($_POST['restrictions'] ?? [] as $restriction_values) {
+   // Split fkey and ids
+   $restriction_values = explode("-", $restriction);
+   $fkey = $restriction_values[0];
+   $value = $restriction_values[1];
+
+   // Parse values
+   if ($fkey == User::getForeignKeyField()) {
+      $input['users'][] = $value;
+   } elseif ($fkey == Group::getForeignKeyField()) {
+      $input['groups'][] = $value;
+   } elseif ($fkey == Profile::getForeignKeyField()) {
+      $input['profiles'][] = $value;
+   }
+}
```

Comment:
move this in a method in PluginFormcreatorCommon to make it unit testable avoid to have logic in front/*

By the way, this would be even better in GLPI core as you probably used the new GLPI component to select access rights.

Reply:
I'm looking into if this can be done in core, will update when done.

Reply:
Done 87a28646.